tests/check/check_h26xsigned_auth.c

```diff
@@ -1689,6 +1689,238 @@ START_TEST(vendor_axis_communications_operation)
 END_TEST
 #endif
 
+static signed_video_t *
+generate_and_set_private_key_on_camera_side(struct sv_setting setting, bool add_public_key_to_sei, nalu_list_item_t **sei)
+{
+  SignedVideoReturnCode sv_rc;
+  char *private_key = NULL;
+  size_t private_key_size = 0;
+  signed_video_nalu_to_prepend_t nalu_to_prepend = {0};
+  nalu_list_item_t *i_nalu = nalu_list_item_create_and_set_id("I", 0, setting.codec);
+
+  signed_video_t *sv = signed_video_create(setting.codec);
+  ck_assert(sv);
+  // Read and set content of private_key.
+  sv_rc = signed_video_generate_private_key(setting.algo, "./", &private_key, &private_key_size);
+  ck_assert_int_eq(sv_rc, SV_OK);
+  sv_rc = signed_video_set_private_key(sv, setting.algo, private_key, private_key_size);
+  ck_assert_int_eq(sv_rc, SV_OK);
+
+  sv_rc = signed_video_add_public_key_to_sei(sv, add_public_key_to_sei);
+  ck_assert_int_eq(sv_rc, SV_OK);
+  sv_rc = signed_video_set_authenticity_level(sv, setting.auth_level);
+  ck_assert_int_eq(sv_rc, SV_OK);
+
+  // Add an I-NALU to trigger a SEI.
+  sv_rc = signed_video_add_nalu_for_signing(sv, i_nalu->data, i_nalu->data_size);
+  ck_assert_int_eq(sv_rc, SV_OK);
+  sv_rc = signed_video_get_nalu_to_prepend(sv, &nalu_to_prepend);
+  ck_assert_int_eq(sv_rc, SV_OK);
+  *sei = nalu_list_create_item(nalu_to_prepend.nalu_data, nalu_to_prepend.nalu_data_size,
+      setting.codec);
+  sv_rc = signed_video_get_nalu_to_prepend(sv, &nalu_to_prepend);
+  ck_assert_int_eq(sv_rc, SV_OK);
+  ck_assert(nalu_to_prepend.prepend_instruction == SIGNED_VIDEO_PREPEND_NOTHING);
+  ck_assert(tag_is_present(*sei, setting.codec, PUBLIC_KEY_TAG) == add_public_key_to_sei);
+  free(private_key);
+
+  return sv;
+}
+
+static void validate_public_key_scenario(signed_video_t *sv, nalu_list_item_t *sei, bool wrong_key,
+    signature_info_t *signature_info, SignedVideoCodec codec)
+{
+  SignedVideoReturnCode sv_rc;
+  bool public_key_present = sv->has_public_key;
+
+  // Validate this first GOP.
+  signed_video_authenticity_t *auth_report = NULL;
+  signed_video_latest_validation_t *latest = NULL;
+
+  nalu_list_item_t *i_nalu = nalu_list_item_create_and_set_id("I", 0, codec);
+  sv_rc = signed_video_add_nalu_and_authenticate(sv, sei->data, sei->data_size, &auth_report);
+  ck_assert_int_eq(sv_rc, SV_OK);
+  ck_assert(!auth_report);
+
+  // Late public key
+  if (signature_info) {
+    sv_rc = signed_video_set_public_key(
+        sv, signature_info->public_key, signature_info->public_key_size);
+    ck_assert_int_eq(sv_rc, SV_NOT_SUPPORTED);
```

Comment:
It would be valuable with a comment that since setting a public key after the session start is not supported, there is no point in adding the i_nalu and authenticate.